core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+import scala.util.matching.Regex
+
+object AutoBump {
+  sealed trait Warnings {
+    def warn(log: Logger): IO[Unit]
+  }
+  case object UpdateError extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn("was unable to run `sbt update` following the trickle application")
+      log.warn("this may mean that the some artifacts are not yet propagated; skipping")
+    }
+  }
+  case object NoChangesError extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn("git-commit exited with error")
+      log.warn("this usually means the target repository was *already* at the latest version but hasn't published yet")
+      log.warn("you should check for a stuck trickle PR on that repository")
+    }
+  }
+  case object PushError extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn("git-push exited with error")
+      log.warn("this usually means some other repository updated the pull request before this one")
+    }
+  }
+  case class NotOldest(oldest: PullRequestDraft, draft: PullRequestDraft) extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn(s"pull request ${draft.number} is newer than existing pull request ${oldest.number}")
+      log.warn("this usually means two or more repositories finished build at the same time,")
+      log.warn("and some other repository beat this one to pull request creation.")
+    }
+  }
+
+  sealed abstract class ChangeLabel(val label: String)
+  object ChangeLabel {
+    val values: List[ChangeLabel] = List(RevisionChange, FeatureChange, BreakingChange)
+    val order: Map[ChangeLabel, Int] = values.zipWithIndex.toMap
+    implicit val ordering: Ordering[ChangeLabel] = Ordering.by(order)
+    val fromString: Map[String, ChangeLabel] = values.map(change => change.label -> change).toMap
+    val labelPattern: Regex = values.mkString("|").r
+
+    def apply(label: String): Option[ChangeLabel] = {
+      fromString.get(label)
+    }
+
+    def unapply(arg: String): Option[ChangeLabel] = labelPattern.findFirstIn(arg).flatMap(apply)
+  }
+  case object RevisionChange extends ChangeLabel("version: revision")
+  case object FeatureChange extends ChangeLabel("version: feature")
+  case object BreakingChange extends ChangeLabel("version: breaking")
+
+  implicit class Unpack[F[_]: Monad, A](response: F[GHResponse[A]]) {
+    def unpack: F[Either[GHException, A]] = EitherT(response).map(_.result).value
+  }
+
+  val AutoBumpLabel = ":robot:"
+  val PullRequestFilters: List[PRFilter] = List(PRFilterOpen, PRFilterSortCreated, PRFilterOrderAsc, PRFilterBase("master"))
+  val LinkRelation: Regex = """<(.*?)>; rel="(\w+)"""".r
+  val LabelPattern: Regex = "version: \\w+".r
+
+  def autoBumpCommitTitle(author: String) = s"Applied dependency updates by $author"
+
+  def autoPage[F[_]: Sync, T](first: Pagination)
+                             (call: Pagination => F[Either[GHException, GHResult[List[T]]]]): Stream[F, T] = {
```

Comment:
Style nit! This is incorrectly formatted. Actually, most of your long signatures are.

We generally wrap signatures by wrapping all blocks to a double-indented context, *including* the return type, which we wrap before the `:`. Thus, your signature should be the following:

```suggestion
  def autoPage[F[_]: Sync, T](
      first: Pagination)(
      call: Pagination => F[Either[GHException, GHResult[List[T]]]])
      : Stream[F, T] = {
```

Basically as a rule of thumb, if you're vertically aligning *anything* other than sbt's dependency `%` operators, it's not conventional.

Reply:
Ugh, I hate dangling return types. :(

Reply:
If we had a scalastyle configs, IntelliJ would have pointed that stuff out to me.